### PR TITLE
save and load templates, with tabula-java-1.0.0

### PR DESCRIPTION
**Pull request by jeremybmerrill**
this set of commits adds UI and serverside ability to

save a template from the extraction view
load an already-saved template from the extraction view
upload/download/rename a template from disk in the Library (front page) view
this incorporates @cheapsteak's #669 PR. this will close the feature requested in #608, #483, #93, #505.

The "template" is the same as the JSON output you can already download from the export view. Template metadata is recorded in workspace.json, which now has a new format (with stuff to automatically convert existing workspace.json files).

@jazzido @mtigas, anyone else, wanna take a look? (I already showed Mike IRL)

this is kind of just a first-stab at the problem, there's a bunch of enhancements that could be done now:

refactor tabula_web.rb's template persistence stuff, which is a little repetitive and gross right now
reimplement Autodetected Selections as a template, since it's really the same thing
how should we handle repeated selections? (i.e. if the template is saved with one selection repeated on pages 2 thru 100 of 100 page PDF, and that selection is loaded on a 102 page PDF, should we apply the selection to the following pages?)
you should be able to name templates from PDFView
you should be able to "overwrite" old templates from PDFView (i.e. loading a template, modifying it, and saving it as teh same one, not a new one)
maybe you should be able to upload templates in PDFView (rather than just on the front page)
"batch" mode from front page, applying one template to multiple PDFs all at once.
how to handle selections that don't "match" the PDF they're applied to -- maybe a selection's coordinates are outside the page on the new PDF, or there are selections on pages that don't exist in the new PDF.

This PR is an updated version of #705, except it's proposed to merge into tabula-java-1.0.0 per @jazzido's request, so we can do it all at once.

### CODE REVIEW

**Comment by jeremybmerrill** [comment]
@jazzido I'll also note: this converts from the old workspace.json format to the new one. The new singleton Tabula::Workspace thingy works great, thanks for adding that!

File: webapp/tabula_web.rb [hunk 1]
```diff
@@ -77,6 +77,62 @@ def upload(file)
   return [job_batch, file_id]
 end
 
+
+def create_template(template_info)
+  puts template_info.inspect
+  template_name = template_info["name"] || "Unnamed Template #{Time.now.to_s}"
```
Comment (jazzido) [inline webapp/tabula_web.rb]: @jeremybmerrill can we move this to `Tabula::Workspace`? Thanks!

**Comment by jazzido** [comment]
Awesome, @jeremybmerrill ! Thanks a lot. Just submitted a minor change request.

File: webapp/tabula_web.rb (hunk 1 above)
Comment (jeremybmerrill) [inline webapp/tabula_web.rb]: I inlined it in the route where it's called. I think it's not really about manipulating the workspace, but instead about manipulating templates. https://github.com/tabulapdf/tabula/pull/711/commits/e9fd39474e0cdb19c096eca7308c37e50ecdd6ce

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ language: ruby
 before_install:
   - gem update --system
   - gem install bundler
+  - gem install jbundler
```
Comment (jazzido) [inline .travis.yml]: Man, Travis is driving me nuts. `jbundler` is in the `Gemfile`, so that wouldn't be necessary right? So frustrating.

**Comment by jazzido** [comment]
BTW, it'd be nice to give some feedback when the user clicks the "Save as template" button.

File: .travis.yml (hunk 2 above)
Comment (jeremybmerrill) [inline .travis.yml]: Soooo frustrating, yes.

**Comment by jazzido** [comment]
OMG it worked. Thanks!

**Comment by Alamin-pro** [comment]
please how can i c reate a template.json

**Pull request merged**